src/odemis/driver/tmcm.py

```diff
@@ -843,11 +843,24 @@ def SendInstruction(self, n, typ=0, mot=0, val=0):
         msg[-1] = numpy.sum(msg[:-1], dtype=numpy.uint8)
         with self._ser_access:
             logging.debug("Sending %s", self._instr_to_str(msg))
-            self._serial.write(msg)
+            try:
+                self._serial.write(msg)
+            except IOError:
+                logging.warn("Failed to send command to TMCM, trying to reconnect.")
+                self._tryRecover()
+                # send instruction again
```

Comment:
I'd suggest to make the comment more explicit: "Failure here should mean that the device didn't get the (complete) instruction, so it's safe to send the command again."